x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/create_connector_flyout/index.tsx

```diff
@@ -159,6 +168,11 @@ const CreateConnectorFlyoutComponent: React.FC<CreateConnectorFlyoutProps> = ({
 
       const createdConnector = await createConnector(validConnector);
       return createdConnector;
+    } else {
```

Comment:
> If a form has **only** required fields (no optional fields)

Having the code here will not meet this requirement. As is the message is triggered whenever the form is not valid.

We should not display a message saying **all** fields are required when some are optional.

<img width="851" alt="Screenshot 2024-12-04 at 10 36 09" src="https://github.com/user-attachments/assets/c0b8a25f-272d-4be4-bf57-375d767e738b">

This change will affect all connectors.

Reply:
Should I restrict this change only to `gen AI` connectors that we use in `playground` as most of these have required fields, or should all connectors have the same behavior?

cc: @mdefazio

Reply:
@adcoelho What if we make the message more generic? 

"There are required fields missing" or "Some fields are not valid" 

Ideally we will have some message at the top letting the user know there are errors in the form. (which I think is the point of the PR?)

Reply:
> What if we make the message more generic?
> "There are required fields missing" or "Some fields are not valid"

These would imo make more sense but I am also trying to be cautious here about a UI change that would affect **all** connectors. We didn't discuss it internally and weren't aware. But it doesn't seem like a big deal, I'll bring it up in our sync(today).

Reply:
> > What if we make the message more generic?
> > "There are required fields missing" or "Some fields are not valid"
> 
> These would imo make more sense but I am also trying to be cautious here about a UI change that would affect **all** connectors. We didn't discuss it internally and weren't aware. But it doesn't seem like a big deal, I'll bring it up in our sync(today).

Thank you @adcoelho. Please keep us informed about the team's decision.

Reply:
Ok, getting back to this @mdefazio 

> What if we make the message more generic?
> "There are required fields missing" or "Some fields are not valid"

Won't your suggestion then go against [the original comment](https://github.com/elastic/kibana/issues/195048#issuecomment-2393788073) from @MichaelMarcialis that originated this task and where he explicitly says:

> If the form has one or more optional fields, no additional affordance is necessary, beyond marking the optional fields.

?

This change only seems to make sense if it applies to connectors where **all fields are required** and if not we should leave things as is.

Reply:
The general idea is that if there are any errors in the form, a header should appear at the top with a message like, `There are some errors while submitting`. Individual fields or actions can then display their specific error messages for more details. However, I'll let @mdefazio confirm.

Reply:
Correct—this is about errors in the form. We removed the message "All fields required" that appears initially prior to submission since this wouldn't make sense for all Connector forms. 

Related, the original comment is around noting which fields are required, prior to form submission, not about error handling. 

If i'm not mistaken, the benefit of this additional callout, is to indicate that there are errors in the form, which may be out of view initially, and provide a landmark and message to screen readers. 

If we make this message generic, meaning simply "There are fields with errors.", or "Some fields were not valid", etc., it allows us to show this regardless if all fields are required or not—so for every connector type form.

Someone can correct me on the need and benefit of this, but that is my understanding.  I acknowledge that the Connector framework is unique based on the dynamic nature of the forms, so this is a tricky one to start off with for this pattern.

Reply:
FWIW, my comment wasn't saying that an error callout such a this can't be shown on form submission. I was just stating that a simple note stating that "All items are required" (shown before form submission) isn't to be used on a form that has one or more optional elements. If ya'll feel this form is lengthy, with some elements with validation errors falling out of view, and could benefit from the inclusion of an additional callout indicating errors, I'll leave that to your discretion.